### DISCUSSION TITLE
cron supervisorctl issue

### DISCUSSION THREAD

**Discussion by ping38** [lloesche/valheim-server-docker#236] [Q&A]
I get below error (and nos server restart) at cron/scheduled server restart time (5h10 as expected): any idea ?
Mar 17 05:10:00 417585c4330d cron.info crond[33]: USER root pid 18079 cmd /usr/bin/supervisorctl restart valheim-server
/bin/bash: /usr/bin/supervisorctl: No such file or directory

**Comment by ping38** [comment]
issue should be fixed with #216 
I'm probably working with an old docker release.
I've in mind that systemd restart will autopull/update valheim-server-docker:main container release ?

[Service]
TimeoutStartSec=0
ExecStartPre=-/usr/bin/docker stop %n
ExecStartPre=-/usr/bin/docker rm %n
ExecStartPre=/usr/bin/docker pull **lloesche/valheim-server:main**

**Reply by lloesche** [reply]
Looks correct. What if you manually pull? Does it update the image?

**Reply by ping38** [reply]
Confirmed: the docker images (main) was from last week so #216 should now be fixed the main issue.
I've made a BIG manual cleanup in the image local repo, when manualy pulled it's OK.
But I've the feeling that systemd ExecStartPre is not working as expected ... under surveillance

**Reply by lloesche** [reply]
The latest version of the systemd service unit no longer uses the Pre command and just runs docker with `--pull=always`.

**Reply by ping38** [reply]
Thx it works perfectly
